file/writable_file_writer.cc

```diff
@@ -487,8 +504,17 @@ IOStatus WritableFileWriter::WriteDirect() {
         start_ts = FileOperationInfo::StartNow();
       }
       // direct writes must be positional
-      s = writable_file_->PositionedAppend(Slice(src, size), write_offset,
-                                           IOOptions(), nullptr);
+      if (perform_data_verification_) {
+        std::string checksum;
```

Comment:
Building on Siying's previous comments about efficiency: I don't think we should do a heap allocation (std::string append) on every file write with handoff checksum.

Reply:
I replaced it with char checksum_buf[sizeof(uint32_t)], might be a better way to avoid string append.